explorer/interpreter/type_checker.cpp

```diff
@@ -671,7 +697,21 @@ auto TypeChecker::ImplicitlyConvert(std::string_view context,
     }
 
     // Perform the builtin conversion.
-    return arena_->New<BuiltinConvertExpression>(source, destination);
+    auto* convert_expr =
+        arena_->New<BuiltinConvertExpression>(source, destination);
+
+    // For subtyping, rewritte into successive `.base` accesses.
+    if (isa<PointerType>(source_type) && isa<PointerType>(destination) &&
+        cast<PointerType>(destination)->pointee_type().kind() ==
+            Value::Kind::NominalClassType) {
+      CARBON_ASSIGN_OR_RETURN(
+          const auto* conv_expr,
+          BuildSubtypeConversion(source, cast<PointerType>(source_type),
+                                 cast<PointerType>(destination)))
+      convert_expr->set_rewritten_form(conv_expr);
```

Comment:
```suggestion
          const auto* rewrite,
          BuildSubtypeConversion(source, cast<PointerType>(source_type),
                                 cast<PointerType>(destination)))
      convert_expr->set_rewritten_form(rewrite);
```

It bugs me to have both convert_expr and conv_expr. :)

Reply:
Good catch, agree.